paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -547,6 +549,18 @@ private void validateCustomTablePath(Map<String, String> options) {
         }
     }
 
+    private List<SchemaChange> filterNonEmptyKeyInSchemaChange(List<SchemaChange> changes) {
+        List<SchemaChange> schemaChanges = new ArrayList<>();
+        for (SchemaChange schemaChange : changes) {
+            if (schemaChange instanceof SchemaChange.SetOption
+                    && StringUtils.isEmpty(((SchemaChange.SetOption) schemaChange).key())) {
```

Comment:
What about throw a exception instead of continue?

Reply:
Thanks, empty is a non-sense config, maybe filter directly also OK.